pkg/config/application.go

```diff
@@ -644,3 +646,25 @@ func (c *DeploymentChainTriggerCondition) Validate() error {
 	}
 	return nil
 }
+
+func LoadApplication(repoPath, configRelPath string, appKind model.ApplicationKind) (*GenericApplicationSpec, error) {
+	var absPath = filepath.Join(repoPath, configRelPath)
+
+	cfg, err := LoadFromYAML(absPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("application config file %s was not found in Git", configRelPath)
+		}
+		return nil, err
+	}
+	if kind, ok := ToApplicationKind(cfg.Kind); !ok || kind != appKind {
```

Comment:
just curious about one thing, but can we use this?
https://github.com/pipe-cd/pipecd/blob/master/pkg/config/config.go#L73

Reply:
to me, these 2 functions look simular and look like one of them is redundant
https://github.com/pipe-cd/pipecd/blob/master/pkg/config/config.go#L73

Reply:
Nice catch!
Let's remove the `ToApplicationKind` function and use the `ToApplicationKind` method instead of that one.